app/lib/rocdev/slack/slack_api.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Rocdev.Slack.SlackAPI do
+  @moduledoc """
+  Provides access to Slack Invitation API.
+  """
+  use HTTPoison.Base
+  alias Rocdev.Slack.SlackAPIHelper
+
+  @base_url Application.get_env(:rocdev, :slack_api_base_url)
+  @token Application.get_env(:rocdev, :slack_api_token)
+
+  def team do
+    get("/team.info")
+    |> SlackAPIHelper.handle_response
```

Comment:
@dantswain is a helper like this idiomatic or is there a better alternative?

Reply:
Blech.  I feel like this is the `use` pattern leaking backwards into your code.  Without thinking about it _too_ much, I suggest something like this

*  Write a module like `SlackEndpoint` that just wraps get/post/etc - i.e., `SlackEndpoint.get(path, params)` does the HTTP GET and returns `{:ok, <parsed json>}` or `{:error, <error description>}`.
* Write a module like `SlackAPI` that uses `SlackEndpoint` to do the work.

That wall off all of this gross stuff inside `SlackEndpoint`.

If you want to go totally FP you could try to make it something that does function composition.

* `SlackAPI.Request` turns business logic into struct objects.  The struct would encode the request type (GET/POST/etc), the endpoint path, and the payload.
* `SlackAPI.Response` would know how to interpret the HTTP responses
* `SlackAPI.Endpoint` performs HTTP requests given `SlackAPI.Request` and returns `SlackAPI.Response`
* `SlackAPI` wraps it all up

So you would end up with something like

```
defmodule SlackAPI do
   alias SlackAPI.Endpoint
   alias SlackAPI.Request
   alias SlackAPI.Response

  # optional DI to support testing
  def team(endpoint \\ Endpoint) do
    Request.get("/team.info")
    |> endpoint.perform_request
    |> Response.payload_data("whatever")
  end
end
```